whois-nrtm4/src/main/java/net/ripe/db/nrtm4/util/SnapshotUpdateWindow.java

```diff
@@ -0,0 +1,39 @@
+package net.ripe.db.nrtm4.util;
+
+import net.ripe.db.nrtm4.persist.SnapshotFile;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+@Component
+public class SnapshotUpdateWindow {
+
+    private final Date startTime;
+    private final Date endTime;
+
+    SnapshotUpdateWindow(@Value("${nrtm.snapshot.window:00:00-04:00}") final String snapshotWindowString) {
```

Comment:
Can we hard-code a schedule for generating a snapshot? e.g. run the update job every minute and use logic if enough time has passed or deltas generated to create a new snapshot.

Reply:
that's pretty much the idea behind this code. it's not finished yet -- i'll remove this until we have a nice way to do it.